frontend/main.js

```diff
@@ -127,21 +146,41 @@ async function startApp () {
         }
       }
     },
-    mounted () {
+    async mounted () {
+      sbp('okTurtles.data/set', PUBSUB_INSTANCE, sbp('chelonia/connect'))
       const reducedMotionQuery = window.matchMedia('(prefers-reduced-motion: reduce)') || {}
       if (reducedMotionQuery.matches || this.isInCypress) {
         this.setReducedMotion(true)
       }
+      const { bannerGeneral } = this.$refs
+      sbp('okTurtles.data/set', 'BANNER', bannerGeneral) // make it globally accessible
+      // display a self-clearing banner that shows up after we've taken 2 or more seconds
+      // to sync a contract.
+      let clearBannerTimer
+      const debouncedSyncBanner = debounce((cID) => {
+        const clearBanner = () => {
+          if (!this.ephemeral.syncs.length) {
+            if (bannerGeneral.severity() !== 'danger') {
+              bannerGeneral.clean()
+            }
+            clearBannerTimer = undefined
+          } else {
+            // after first 2 seconds, check every half a second for sync to finish
+            setTimeout(clearBanner, 500)
+          }
+        }
+        if (!clearBannerTimer && this.ephemeral.syncs.length) {
+          bannerGeneral.show(L('Loading events from server...'), 'wifi')
+          clearBannerTimer = setTimeout(clearBanner, 2000)
+        }
+      }, 2000)
```

Comment:
I was working on that issue too. Checking whether the banner has already a higher severity status is a great idea!

For my part, my solution filters out contract IDs from `state.pending`, to avoid triggering such banner activity when signing up or creating a new group.

Reply:
In the related issue however, the delays were differents, like "show the banner after 1s then don't hide it again before another 2s"

Reply:
Also I feel like more of this banner-related code should be moved to `BannerGeneral.vue` rather than stay in `main.js`.

Reply:
> In the related issue however, the delays were different, like "show the banner after 1s then don't hide it again before another 2s"

Yeah, I edited the issue because I decided they should be slightly tweaked. 1 second to read a message seemed a bit short to me.

> Also I feel like more of this banner-related code should be moved to BannerGeneral.vue rather than stay in main.js.

Good point, I'll add a new `debouncedShow` method to BannerGeneral.vue and will push